### PR TITLE
 [ALLUXIO-3207] Journal backup and restore

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-3207

The main change needed to merge to master was moving journal backup to `MetaMaster` now that there is a real `MetaMaster`. The restore path needed to stay in `AlluxioMasterProcess` because the journal needs to be initialized before starting sub-masters. To maintain the coordination between backup and restore, I moved the logic to a new `BackupManager` which is a part of `MasterContext`. I also added a utility method to make it easier to work with `MasterContext` from tests.

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by aaudiber** [comment]
Jenkins, test this please

File: bin/alluxio-start.sh [hunk 1]
```diff
@@ -43,11 +43,12 @@ MOPT (Mount Option) is one of:
...
   SudoMount is assumed if MOPT is not specified.
 
--f  format Journal, UnderFS Data and Workers Folder on master
--N  do not try to kill previous running processes before starting new ones
--w  wait for processes to end before returning
--m  launch monitor process to ensure the target processes come up.
--h  display this help."
+-f         format Journal, UnderFS Data and Workers Folder on master.
+-h         display this help.
+-i backup  a journal backup to restore the master from.
```
Comment (gpang) [inline bin/alluxio-start.sh]: can you mention the format of this backup, and/or examples?

Comment (aaudiber) [reply bin/alluxio-start.sh]: good idea, done

**Pull request merged**